appservice/src/SiteWrapper.ts

```diff
@@ -116,7 +126,60 @@ export class SiteWrapper {
         this.log(outputChannel, 'Deployment completed.');
     }
 
-    private async waitForDeploymentToComplete(kuduClient: KuduClient, outputChannel: vscode.OutputChannel, pollingInterval: number = 5000): Promise<void> {
+    public async localGitDeploy(fsPath: string, client: WebSiteManagementClient, outputChannel: vscode.OutputChannel, servicePlan: string): Promise<DeployResult | undefined> {
```

Comment:
I don't think we should have servicePlan as a parameter since it's only used for telemetry purposes. The extension should take care of telemetry, not the npm package

Reply:
I wouldn't even include `LocalGitDeployError` in this package

Reply:
How do we distinguish what caused the error if all errors return undefined?